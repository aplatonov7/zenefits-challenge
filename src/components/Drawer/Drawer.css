.Drawer__container {
  position: absolute;
  left: 0;
  top: 0;
  width: 450px;
  max-width: 100%;
  transform: translateX(-100%);
  transition-property: -webkit-transform,transform,opacity;
  transition-duration: 0.2s;
  transition-timing-function: cubic-bezier(0.0,0.0,0.2,1);
  z-index: 1;
  display: flex;
  flex-direction: column;
}

.Drawer__container.opened {
  transform: translateX(0);
}

.Drawer__container.full-height {
  height: 100%;
}

.Drawer__button {
  position: absolute;
  top: 0.5rem;
  right: -1.5rem;
  width: 1.5rem;
  height: 3.5rem;
  box-shadow: 2px 1px 2px rgba(0, 0, 0, 0.3);
  background-color: #fff;
  display: flex;
  align-content: center;
  justify-content: center;
  font-size: 0;
  cursor: pointer;
  z-index: 1;
}

.Drawer__button:after {
  content: '\203A';
  font-size: 2rem;
  color: #555;
}

.Drawer__container.opened .Drawer__button:after {
  content: '\2039';
}

.Drawer__anchor {
  position: absolute;
  right: -40px;
  height: 100%;
  width: 40px;
}