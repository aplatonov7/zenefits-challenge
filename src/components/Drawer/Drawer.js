import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import Swipeable from 'react-swipeable'

import './Drawer.css'

const Drawer = ({ children, showToggleButton, opened, setOpened }) => (
  <Swipeable className={cx('Drawer__container', {
    'opened': opened,
    'full-height': showToggleButton,
  })} onSwipedRight={() => setOpened(true)} onSwipedLeft={() => setOpened(false)}>
    {children}
    {showToggleButton && (
      <button className="Drawer__button" onClick={() => setOpened(!opened)}>
        Toggle the drawer
      </button>
    )}
    <div className="Drawer__anchor"></div>
  </Swipeable>
)

Drawer.propTypes = {
  children: PropTypes.node,
  opened: PropTypes.bool,
  showToggleButton: PropTypes.bool,
  setOpened: PropTypes.func,
}

export default Drawer