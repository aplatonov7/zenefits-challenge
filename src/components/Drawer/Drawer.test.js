import React from 'react'

import Drawer from './Drawer'

describe('Drawer component', () => {
  let component

  beforeEach(() => {
    component = shallow(<Drawer />)
  })

  it('Should match snapshot', () => {
    const props = {
      opened: true,
      showToggleButton: true,
      children: 'Inner content',
      setOpened: jest.fn(), 
    }
    expect(renderer.create(<Drawer {...props} />).toJSON()).toMatchSnapshot()
  })

  it('Should show and hide button based on showToggleButton prop', () => {
    component.setProps({ showToggleButton: true })
    expect(component.find('button.Drawer__button')).toHaveLength(1)
    component.setProps({ showToggleButton: false })
    expect(component.find('button.Drawer__button')).toHaveLength(0)
  })

  it('Should add .opened class to the container based on opened prop', () => {
    component.setProps({ opened: false })
    expect(component.hasClass('opened')).toBe(false)
    component.setProps({ opened: true })
    expect(component.hasClass('opened')).toBe(true)
  })

  it('Should fire setOpened prop on toggle button click', () => {
    const mockFn = jest.fn()
    component.setProps({ showToggleButton: true, setOpened: mockFn })
    component.find('button.Drawer__button').simulate('click')
    expect(mockFn).toBeCalled()
  })
  
  it('Should render children', () => {
    component.setProps({ children: 'Inner content' })
    expect(component.contains('Inner content')).toBe(true)
  })
})