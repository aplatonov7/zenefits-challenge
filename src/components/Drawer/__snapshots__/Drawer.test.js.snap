// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer component Should match snapshot 1`] = `
<div
  className="Drawer__container opened full-height"
  onMouseDown={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
>
  Inner content
  <button
    className="Drawer__button"
    onClick={[Function]}
  >
    Toggle the drawer
  </button>
  <div
    className="Drawer__anchor"
  />
</div>
`;
